File: E-EstateV2/src/app/_interface/company.ts
import { Estate } from "./estate";
import { Ownership } from "./ownership";
import { Town } from "./town";

export interface Company {
    id: number
    companyName: string
    address1: string
    address2: string
    address3: string
    postcode: string
    towns: Town[]
    townId: number
    state: string
    stateId: number
    ownerships:Ownership[]
    ownershipId:number
    phone: string
    email: string
    fax: string
    contactNo: string
    managerName: string
    isActive: boolean
    estates: Estate[]
    townEstate: string
    membershipType: string
    createdBy: string
    createdDate: Date
    updatedBy: string
    updatedDate: Date
}

File: E-EstateV2 API/E-EstateV2 API/Models/Company.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace E_EstateV2_API.Models
{
    public class Company
    {
        [Key]
        public int Id { get; set; }
        public string companyName { get; set; }
        public string address1 { get; set; }
        public string address2 { get; set; }
        public string address3 { get; set; }
        public string postcode { get; set; }
        public string phone { get; set; }
        public string email { get; set; }
        public string fax { get; set; }
        public string contactNo { get; set; }
        public string managerName { get; set; }
        public bool isActive { get; set; }
        public string createdBy { get; set; }
        public DateTime createdDate { get; set; }
        public string updatedBy { get; set; }
        public DateTime updatedDate { get; set; }

        [ForeignKey("TownId")]
        public int townId { get; set; }
        public Town Town { get; set; }

        [ForeignKey("OwnershipId")]
        public int ownershipId { get; set; }
        public Ownership Ownership { get; set; }

        //many to many relation
        public ICollection<BuyerCompany> BuyerCompanies { get; set; }
        public ICollection<SellerCompany> SellerCompanies { get; set; }

    }
}
